easybuild/easyblocks/q/quantumespresso.py

```diff
@@ -73,16 +73,78 @@ def patch_step(self):
     def configure_step(self):
         """Custom configuration procedure for Quantum ESPRESSO."""
 
+        # compose list of DFLAGS (flag, value, keep_stuff)
+        # for guidelines, see include/defs.h.README in sources
+        dflags = []
+
+        repls = []
+
+        extra_libs = []
+
+        comp_fam_dflags = {
+            toolchain.INTELCOMP: '-D__INTEL',
+            toolchain.GCC: '-D__GFORTRAN -D__STD_F95',
+        }
+        dflags.append(comp_fam_dflags[self.toolchain.comp_family()])
```

Comment:
This will result in a hard crash when any other compiler family is used (`KeyError`), that should be handled better to produce a clean error message mentioning the compiler family?

Reply:
Probably, not something i changed, it just moved from line 102
We can target that later i think

Reply:
would a code like:
if not self.toolchain.comp_family() in comp_fam_dflags:
  raise EBerror('text')

work?

Reply:
For example:
        if not self.toolchain.comp_family() in comp_fam_dflags:
            raise EasyBuildError("Toolchain %s is not yet supported in QuantumESPRESSO")

Reply:
perhaps "supported for" is better?

Reply:
@akesandgren Yes, that should work and result in a decent error rather than a `KeyError` + traceback.

```python
comp_fam =  self.toolchain.comp_family()
if comp_fam in comp_fam_dflags:
    dflags.append(comp_fam_dflags[comp_fam])
else:
    raise EasyBuildError("...")